test/test_stateless.py

```diff
@@ -219,11 +228,50 @@ def forward(self, x):
 
         a = {'foo': torch.zeros(())}
         mod = Foo()
-        stateless.functional_call(mod, a, torch.ones(()))
+        functional_call(mod, a, torch.ones(()))
         self.assertEqual(mod.foo, torch.zeros(()))
         self.assertEqual(a['foo'], torch.ones(()))
 
 
+    @parametrize("functional_call", [torch.func.functional_call, stateless.functional_call])
+    def test_functional_call_with_kwargs(self, functional_call):
+        class Foo(torch.nn.Module):
+            def __init__(self, x):
+                super().__init__()
+                self.x = x
+
+            def forward(self, inp, *, other_inp):
+                return inp * self.x + other_inp
+
+        a = {'x': torch.zeros(2, 3)}
+        mod = Foo(torch.randn(2, 3))
+        inp, other_inp = torch.randn(2, 3), torch.randn(2, 3)
+        with self.assertRaisesRegex(TypeError, "missing 1 required keyword-only argument: 'other_inp'"):
+            functional_call(mod, a, inp)
+        res = functional_call(mod, a, inp, {'other_inp': other_inp})
+        self.assertEqual(res, other_inp)
+        res_1 = functional_call(mod, a, (), {'inp': inp, 'other_inp': other_inp})
+        self.assertEqual(res, res_1)
+
+
+    def test_functional_call_multiple_dicts(self):
+        mod = MockModule()
+        x = torch.rand((1, 1))
+        parameters = {k: torch.ones_like(v) for k, v in mod.named_parameters()}
+        buffers = {k: torch.zeros_like(v) for k, v in mod.named_buffers()}
+        res = torch.func.functional_call(mod, (parameters, buffers), x)
+        self.assertEqual(res, x + 1)
```

Comment:
nit:
- would try one example with 3 or more dicts, and another one with zero dicts. I can see a situation in the future where someone just assumes the number of dicts goes up to 2

Reply:
added!